### DISCUSSION TITLE
How to register edge templates (layouts / components / partials) in an npm package

### DISCUSSION THREAD

**Discussion by bahdcoder** [adonisjs/core#2287] [General]
Hello. Would love to find out if there's a possibility to register edge templates in a third-party package, that way if you install the package, you can use a whole bunch of pre-registered templates. 

Let's say I created a package called `adonis-mailer-bootstrap-template`, and anyone using my template can create a simple button using `@!component('button', title = "Reset your password")`. This would be awesome. They can also use layouts registered by my package. 

I tried using `registerTemplate`, but it has no support for slots (i think)

Thanks !

**Comment by thetutlage** [comment]
@bahdcoder Yes, you can do it using the disks of Edge. Lets say you store the templates inside the `node_modules/adonis-mailer-bootstrap-template/views` directory.

The user of your package can register a named disk as

```ts
import View from '@ioc:Adonis/Core/View'

View.disk('bootstrap', absolutePathToThePackageViewsDir)
```

and then they can use it as 

```edge
@component('bootstrap::template-name')
```
